querydsl-jpa/src/main/java/com/querydsl/jpa/impl/JPAProvider.java

```diff
@@ -50,16 +71,23 @@ private static void addMapping(String className, JPQLTemplates templates) {
 
         templatesByName.put("batoo", BatooTemplates.DEFAULT);
         templatesByName.put("eclipselink", EclipseLinkTemplates.DEFAULT);
-        templatesByName.put("hibernate", HQLTemplates.DEFAULT);
+        templatesByName.put("hibernate", hibernateTemplates);
+        templatesByName.put("hibernate5", Hibernate5Templates.DEFAULT);
         templatesByName.put("openjpa", OpenJPATemplates.DEFAULT);
         templatesByName.put("datanucleus", DataNucleusTemplates.DEFAULT);
     }
 
     public static JPQLTemplates getTemplates(EntityManager em) {
         // detect by delegate
         for (Map.Entry<Class<?>, JPQLTemplates> entry : mappings.entrySet()) {
-            if (entry.getKey().isAssignableFrom(em.getDelegate().getClass())) {
-                return entry.getValue();
+            try {
+                if (em.unwrap(entry.getKey()) != null) {
```

Comment:
`unwrap` is preferred over `getDelegate` (even mentioned in the Javadoc of `getDelegate` ). I am keeping the old method as a fallback in the catch clause.